functions/src/index.ts

```diff
@@ -37,27 +39,33 @@ export const index = client.initIndex(config.algoliaIndexName);
 logs.init();
 
 const handleCreateDocument = async (
-  snapshot: DocumentSnapshot
+  snapshot: DocumentSnapshot,
+  timestamp: Number
 ) => {
   try {
-    const data = extract(snapshot);
+    const data = extract(snapshot, timestamp);
+
+    logs.debug({
+      ...data
+    });
 
     logs.createIndex(snapshot.id, data);
-    await index.saveObjects([ data ]);
+    await index.partialUpdateObject(data);
```

Comment:
```suggestion
    await index.partialUpdateObject(data, { createIfNotExists: true });
```

https://www.algolia.com/doc/api-reference/api-methods/partial-update-objects/?client=javascript#method-param-createifnotexists